Neuro/Minigames/Solvers/UnlockSafeSolver.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections;
+using Neuro.Cursor;
+using UnityEngine;
+
+namespace Neuro.Minigames.Solvers;
+
+[MinigameSolver(typeof(SafeMinigame))]
+public class UnlockSafeSolver : IMinigameSolver<SafeMinigame>, IMinigameOpener
+{
+    public bool ShouldOpenConsole(Console console, PlayerTask task) => true;
+
+    public IEnumerator CompleteMinigame(SafeMinigame minigame)
+    {
+        int[] combo = minigame.combo;
+        Transform tumblerTransform = minigame.Tumbler.transform;
+        Transform spinnerTransform = minigame.Spinner.transform;
+        Vector2 tumblerPosition = tumblerTransform.position;
+        Vector2 spinnerPosition = spinnerTransform.position;
+        Vector2 mousePosition = tumblerPosition;
+        float time = 0f;
+
+        // TODO: Why doesn't this work sometimes??
```

Comment:
@Alexejhero do you think you could help me figure out why this only works like 1/4 of the time? Sometimes it resets the combination after verifying that it worked.

I think it might be related to the cursor becoming very jittery sometimes? The jitter seems to get calmed by removing the `yield return null` from `CoMoveTo()` but I don't know if that breaks any other tasks.

Reply:
The cursor stuttering does indeed seem to be the problem. A sound is played and the code resets if the tumbler is move in the wrong direction by even a pixel, and the sound seems to play on every attempt that fails.

Reply:
Not sure, try using InGameCursor.Instance.CoMoveCircle, you can see an example use in InsertKeysSolver.cs

Reply:
I've discovered why the unlock safe task has been such a pain in the ass. For some reason it spawns 2 solver instances when using `IMinigameSolver<SafeMinigame>, IMinigameOpener`, which was causing the cursor jittering and is also making CoMoveCircle act up too. Using a GenericMinigameSolver<SafeMinigame> doesn't have this issue because one of the solver instances to throw an exception for a null NormalPlayerTask.

Reply:
Even if two different instances are created, only one of them is run. I do not see how this can be the problem.